src/main/resources/test_files/runtime-spark/NestedFLWORSpark4.iq.HOLD

```diff
@@ -0,0 +1,4 @@
+(:JIQS: ShouldRun; Output="1" :)
+for $i in parallelize(1) for $l in (for $i in parallelize(1) return 1)
```

Comment:
This looks like a duplicate of NestedFLWORSpark2.jq that works?

Could we change to

for $l in (for $i in parallelize($i) return 1)

and check that it throws RBST0003? (This error code is untested yet, if I am correct).

Reply:
If i may point out tiny difference:
NestedFLWORSpark2
for $i in parallelize(1) for $l in (_let_ $i := parallelize(1) return 1)
return $i

NestedFLWORSpark4.iq.HOLD 
for $i in parallelize(1) for $l in (_for_ $i in parallelize(1) return 1)
return $i

The local evaluation on the let clause was what made test2 work. When this is a query with 4, it woud cause nesting and fail with a NullPointerException.

Reply:
One more thing: the following is not a valid query.
for $l in (for _$i_ in parallelize(_$i_) return 1)
Replacing $i with $$ also doesn't work as I believe there is no prior expression

Reply:
> One more thing: the following is not a valid query.
> for $l in (for _$i_ in parallelize(_$i_) return 1)
> Replacing $i with $$ also doesn't work as I believe there is no prior expression

Correct, that was only a subset. I mean this query:

for $i in parallelize(1)
for $l in (for $i in parallelize($i) return 1)
return $l

should return RBST0003.

Reply:
> If i may point out tiny difference:
> NestedFLWORSpark2
> for $i in parallelize(1) for $l in (_let_ $i := parallelize(1) return 1)
> return $i
> 
> NestedFLWORSpark4.iq.HOLD
> for $i in parallelize(1) for $l in (_for_ $i in parallelize(1) return 1)
> return $i

Yes, I missed this difference. Thank you!

> The local evaluation on the let clause was what made test2 work. When this is a query with for clause, it woud cause nesting and fail with a NullPointerException.

The expression in the second for does not depend on the outer $i: should it not now work, based on what you have just implemented? (i.e., we can simply activate this test, and create the failing one RBST0003 separately) If it does not work, could it be because the variable "hiding" ($i hides $i) that is not working properly?

Reply:
The test works both with and without variable hiding. You can see it under NestedFLWORSpark4 and NestedFLWORSpark5

Reply:
I have also added a negative test that triggers RBST0003 in file NestedFLWORSparkError1.jq

Reply:
Also added the test case you have given above:
for $i in parallelize(1)
for $l in (for $k in parallelize($i) return $k)
return $l